### PR TITLE
Implement`attachToSession` for devtools package

### PR DESCRIPTION
**Pull request by christian-bromann**
## Proposed changes

Currently it was not possible to attach to a running devtools session. This patch adds this functionality.

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have added proper type definitions for new commands (if appropriate)

## Further comments

closes #6459

### CODE REVIEW

File: packages/webdriverio/src/index.ts [hunk 1]
```diff
@@ -77,9 +77,15 @@ export const remote = async function (params: RemoteOptions, remoteModifier?: Fu
...
-export const attach = function (params: WebDriverTypes.AttachOptions): Browser<'async'> {
+export const attach = function (params: AttachOptions): Browser<'async'> {
     const prototype = getPrototype('browser')
-    return WebDriver.attachToSession(params, undefined, prototype, wrapCommand) as WebdriverIO.Browser
+
+    let automationProtocol = 'webdriver'
+    if (params.options?.automationProtocol) {
+        automationProtocol = params.options?.automationProtocol
+    }
+    const ProtocolDriver = require(automationProtocol).default
```
Comment (erwinheitzman) [inline packages/webdriverio/src/index.ts:87]: use dynamic import?

File: packages/devtools/tests/launcher.test.ts [hunk 2]
```diff
@@ -323,3 +326,15 @@ test('launch Edge without Puppeteer default args (backwards compat)', async () =
...
+
+test('connect to existing browser session', async () => {
+    await launch({
+        browserName: 'edge',
+        'ms:edgeOptions': {
+            debuggerAddress: 'localhost:12345'
+        }
+    })
+    expect(puppeteer.launch).not.toBeCalled()
+    expect((puppeteer.connect as jest.Mock))
```
Comment (erwinheitzman) [inline packages/devtools/tests/launcher.test.ts:338]: Not that important but no need to wrap it when it's already enclosed

File: packages/webdriverio/src/index.ts [hunk 3]
```diff
@@ -53,6 +52,7 @@ export const remote = async function (params: RemoteOptions, remoteModifier?: Fu
     const ProtocolDriver = require(automationProtocol).default
 
     params = Object.assign({}, detectBackend(params), params)
+    params.automationProtocol = automationProtocol
```
Comment (erwinheitzman) [inline packages/webdriverio/src/index.ts:55]: the `automationProtocol` is already passed to `updateCapabilities` in the line below it seems. is this actually needed?

**Review by erwinheitzman — COMMENTED** [review]
Left some questions

File: packages/webdriverio/src/index.ts (hunk 3 above)
Comment (christian-bromann) [reply packages/webdriverio/src/index.ts:55]: yes, we want to explicitly set it to ensure that when attaching to it ends up attaching to the correct protocol. But good point on `updateCapabilities`. I moved the assignment in there.

**Pull request merged**